src/sinks/azure_monitor_logs.rs

```diff
@@ -99,6 +103,15 @@ pub struct AzureMonitorLogsConfig {
     #[serde(default)]
     pub request: TowerRequestConfig,
 
+    /// Event field to store the value used to populate the [`TimeGenerated`][1] column.
+    ///
+    /// If unset, the global `log_schema.timestamp_key` is used. This should only be changed if an
+    /// existing `timestamp` field is present, but shouldn't be used for the `TimeGenerated` column.
```

Comment:
We assumed an Azure user would be familiar with TimeGenerated, or easily discover it.

Reply:
If you have a suggestion to make it more understandable - happy to adjust.

Reply:
Maybe something along these lines, if I didn't bork the accuracy:

```
This option should only be set if a `timestamp` field is present in the incoming events.
```

Part of my confusion is, the bits saying: 

"Event field to store the value used to populate the [`TimeGenerated`][1] column."

and 

"but shouldn't be used for the `TimeGenerated` column."

sound contradictory to me.

Reply:
So we point to a field in the http body (via a header) that Azure uses as the `TimeGenerated` value - this is equivalent to to our internal `timestamp` meaning.

By default we had been using `log_schema.timestamp_key` for this key when we encode an event, so it'd be `timestamp` unless the user had configured it differently.

So what I'm trying to get at here is... if you have a `timestamp` field that shouldn't be used for the `TimeGenerated` (say it's an ingest timestamp or something, rather than the event's timestamp) _and_ if `log_schema.timestamp_key` is still "timestamp" you should use this option to not overwrite that field.

It's definitely hard to word, especially succinctly.

Reply:
Here's my attempt with a longer narrative:

```
Use this option to customize the log field used as [`TimeGenerated`][1] in Azure. 

The setting of `log_schema.timestamp_key`, usually `timestamp`, is used here by default. This field should be used in rare cases where `TimeGenerated` should point to specific log field. For example, use this field to set the log field `source_timestamp` as holding the value that should be used as `TimeGenerated` on the Azure side. 
```

Reply:
Addressed in [23926e1](https://github.com/vectordotdev/vector/pull/16475/commits/23926e1d5af45d21f0a09534adbec7792e36c364)